functions/mqhandlers.go

```diff
@@ -251,6 +254,7 @@ func HostUpdate(client mqtt.Client, msg mqtt.Message) {
 			slog.Warn("error checking version less than, but will proceed with upgrade", "error", err)
 		}
 		slog.Info("upgrading client to server's version", "version", sv)
+		upgMutex.Lock()
```

Comment:
can lead to deadlock, if break is executed

Reply:
ok, good point. Added a commit